src/SprykerEco/Zed/Payone/Business/Payment/Hook/PostSaveHook.php

```diff
@@ -0,0 +1,235 @@
+
+        if ($apiLog) {
+            $redirectUrl = $apiLog->getRedirectUrl();
+
+            if ($redirectUrl !== null) {
+                $checkoutResponse->setIsExternalRedirect(true);
+                $checkoutResponse->setRedirectUrl($redirectUrl);
+            }
+
+            $errorCode = $apiLog->getErrorCode();
+
+            if ($errorCode) {
+                $error = new CheckoutErrorTransfer();
+                $error->setMessage($apiLog->getErrorMessageUser());
+                $error->setErrorCode($errorCode);
+                $checkoutResponse->addError($error);
+                $checkoutResponse->setIsSuccess(false);
+            }
+        }
+
+        return $checkoutResponse;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\CheckoutResponseTransfer $checkoutResponse
+     *
+     * @return \Generated\Shared\Transfer\CheckoutResponseTransfer
+     */
+    protected function checkApiAuthorizationRedirectAndError(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponse): CheckoutResponseTransfer
+    {
+        $paymentEntity = $this->queryContainer->createPaymentById($checkoutResponse->getSaveOrder()->getIdSalesOrder())->findOne();
+        $paymentMethodMapper = $this->paymentMapperReader->getRegisteredPaymentMethodMapper($paymentEntity->getPaymentMethod());
+        $requestContainer = $this->getPostSaveHookRequestContainer($paymentMethodMapper, $paymentEntity, $quoteTransfer);
+
+        if ($paymentEntity->getPaymentMethod() === PayoneApiConstants::PAYMENT_METHOD_KLARNA) {
+            $requestContainer = $this->payoneRequestProductDataMapper->mapProductData($quoteTransfer, $requestContainer);
+        }
+
+        $responseContainer = $this->baseAuthorizeSender->performAuthorizationRequest($paymentEntity, $requestContainer);
+
+        if ($responseContainer->getErrorcode()) {
+            $checkoutErrorTransfer = new CheckoutErrorTransfer();
+            $checkoutErrorTransfer->setMessage($responseContainer->getCustomermessage());
+            $checkoutErrorTransfer->setErrorCode($responseContainer->getErrorcode());
+            $checkoutResponse->addError($checkoutErrorTransfer);
+            $checkoutResponse->setIsSuccess(false);
+
+            return $checkoutResponse;
+        }
+
+        if (!$responseContainer->getRedirecturl()) {
+            return $checkoutResponse;
+        }
+
+        $checkoutResponse->setIsExternalRedirect(true);
+        $checkoutResponse->setRedirectUrl($responseContainer->getRedirecturl());
+
+        return $checkoutResponse;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\PayoneApiLogTransfer $apiLogTransfer
+     * @param string $errorCode
+     *
+     * @return \Generated\Shared\Transfer\CheckoutErrorTransfer
+     */
+    protected function createCheckoutErrorTransfer(PayoneApiLogTransfer $apiLogTransfer, string $errorCode): CheckoutErrorTransfer
```

Comment:
* Change a method to `createCheckoutErrorTransfer(string $errorMessage, string $errorCode)`.
* Add a usage of the method to line 142 and 172.

Reply:
++